catalog/spatial/registry/registry-schema-bindings/src/main/java/org/codice/ddf/registry/schemabindings/helper/MapToSchemaElement.java

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.registry.schemabindings.helper;
+
+import static org.codice.ddf.registry.schemabindings.EbrimConstants.RIM_FACTORY;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.BiConsumer;
+import java.util.function.Supplier;
+
+import org.apache.commons.collections.MapUtils;
+import org.apache.commons.lang.StringUtils;
+
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.InternationalStringType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.VersionInfoType;
+
+public class MapToSchemaElement<T> {
+    private final Supplier<T> objectFactory;
+
+    public static final InternationalStringTypeHelper INTERNATIONAL_STRING_TYPE_HELPER =
+            new InternationalStringTypeHelper();
+
+    public MapToSchemaElement(Supplier<T> objectFactory) {
+        this.objectFactory = objectFactory;
+    }
+
+    public Optional<T> populateBooleanElement(Map<String, Object> map, String mapKey,
+            final Optional<T> referenceElement, BiConsumer<Boolean, Optional<T>> updater) {
+        Optional<T> elementToPopulate = Optional.empty();
+        if (referenceElement.isPresent()) {
+            elementToPopulate = Optional.of(referenceElement.get());
+        }
+
+        Boolean booleanToPopulate = MapUtils.getBoolean(map, mapKey);
+        if (booleanToPopulate != null) {
+            if (!elementToPopulate.isPresent()) {
+                elementToPopulate = Optional.of(objectFactory.get());
+            }
+
+            updater.accept(booleanToPopulate, elementToPopulate);
+        }
+
+        return elementToPopulate;
+    }
+
+    public Optional<T> populateStringElement(Map<String, Object> map, String mapKey,
+            final Optional<T> referenceElement, BiConsumer<String, Optional<T>> updater) {
```

Comment:
We should be able to change the `updater` type to `BiConsumer<String, T> updater` and do `updater.accept(valueToPopulate, elementToPopulate.get()` on line 71. Same comment applies to `populateBooleanElement()`.